### PR TITLE
✅ Add tests for sanity checks to ensure data consistency

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

This adds some tests that ensure that the data returned by the Process Engine core package remains consistent after performing multiple imports.

PR #5

## How can others test the changes?

Run the tests.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: _integration_tests/test/sanity_checks.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    should.exist(existingProcessModel);
+    should(existingProcessModel.id).be.equal(processModelId);
+
+    const startEvent = existingProcessModel.flowNodes.find((flowNode) => {
+      return flowNode.constructor.name === 'StartEvent';
+    });
+
+    // The difference between the two process models is the ID of their StartEvent.
+    // We must be expecting the updated StartEvent ID.
+    const expectedStartEventId = 'StartEvent_666';
```
Comment (robpal94) [inline _integration_tests/test/sanity_checks.js]: 🤘

File: _integration_tests/test/sanity_checks.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    await performImport(processModelXml);
+    await performImport(processModelUpdatedXml);
+  });
+
+  after(async () => {
+    await testFixtureProvider.tearDown();
+  });
+
+  it('should always return the most up to date version of any process definition', async () => {
```
Comment (cmg-dev) [inline _integration_tests/test/sanity_checks.js]: > most up to date

sollte doch eher

> current 

sein, oder?

File: _integration_tests/test/sanity_checks.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    // The difference between the two process models is the ID of their StartEvent.
+    // We must be expecting the updated StartEvent ID.
+    const expectedStartEventId = 'StartEvent_666';
+
+    should.exist(startEvent);
+    should(startEvent.id).be.equal(expectedStartEventId, `Received an unexpected StartEventId: ${startEvent.id}`);
+  });
+
+  it('should not contain outdated versions of any process definitions, when querying all process definitions', async () => {
```
Comment (cmg-dev) [inline _integration_tests/test/sanity_checks.js]: `it('should return only current versions of process definitions, when querying all process definitions', async`

Comment (cmg-dev) [reply _integration_tests/test/sanity_checks.js]: Wenn man das positiv formuliert ist das vielleicht besser.

**Comment by ElRaptorus** [comment]
@cmg-dev done

**Pull request merged**